awx/main/managers.py

```diff
@@ -127,11 +130,21 @@ def register(self, uuid=None, hostname=None, ip_address=None, node_type='hybrid'
                         other_inst.save(update_fields=['ip_address'])
                         logger.warning("IP address {0} conflict detected, ip address unset for host {1}.".format(ip_address, other_hostname))
 
-            # Return existing instance that matches hostname
+            # get the instance based on the hostname
             instance = self.filter(hostname=hostname)
+
+            # Check or update the existing uuid
+            if uuid != None and uuid != self.UUID_DEFAULT:
+                if self.filter(uuid=uuid).exists():
+                    instance = self.filter(uuid=uuid)
```

Comment:
Someone gives both hostname and uuid in the provision_instance command.

There is a hit based on hostname, but there is not a hit based on uuid. First, `instance` is defined to be a non-empty queryset. Then, it is redefined to be an empty queryset.

Later in the code, we find `if instance.exists()` is False and because of that, run `instance = self.create`. The Instance table is still, absolutely, unique on hostname, so it would give an IntegrityError if I read it right.

I could be wrong, but a unit test would be good. I don't know what your expectation is, but that has tie-ins with how the installer is written.